### ISSUE TITLE
[Feature Request] Accessible download links

### ISSUE THREAD

**Issue by darkdragon-001** [mickael-kerjean/filestash#326]
It would be nice, if the `<a href=` attribute would be the download link and the click action/event (`onclick=myFunction()` attribute) would be overwritten by a javascript function to (pre)view. This way, click behavior stays the same, while one additionally use _Copy link address_ to easily download files via tools like `wget`/`curl` which I use regularly.

**Comment by mickael-kerjean** [comment]
I don't understand what you're saying. To my knowledge all the links are already using the a tag:
<img width="1041" alt="Screen Shot 2020-10-12 at 11 21 16 pm" src="https://user-images.githubusercontent.com/13233076/95745958-be9bf980-0ce1-11eb-8bfd-dedfc9952e10.png">

**Comment by darkdragon-001** [comment]
I mean in the overview:
![overview-link-location](https://user-images.githubusercontent.com/177976/95784459-25b2b180-0cd4-11eb-8126-1a3d13a41640.png)

Change it to `/api/files/cat?path=%2FREADME.org`. Your javascript registered an event handler anyways:
![overview-download-location](https://user-images.githubusercontent.com/177976/95784971-23048c00-0cd5-11eb-8d54-12841cbafa70.png)

**Comment by mickael-kerjean** [comment]
I am still not sure to understand. In your screenshot, the handler you've highlighted essentially intercept a click made on a folder to allow for selecting multiple files / folder if you have the [ctrl key pressed](https://github.com/mickael-kerjean/filestash/blob/master/client/pages/filespage/thing-existing.js#L210-L218) to allow for things like this:

![image](https://user-images.githubusercontent.com/13233076/95829885-97493900-0d82-11eb-8977-996ee36a74e5.png)

**Comment by darkdragon-001** [comment]
I don't have a lot of experience with React, probably the framework automatically creates event handlers based on the href attribute. Could you try to set the `href` to the download link and extend your linked click handler to navigate to the preview page when ctrl is _not_ clicked?

**Comment by mickael-kerjean** [comment]
> probably the framework automatically creates event handlers based on the href attribute.

No it doesn't do that, I made that handler on purpose to accommodate the use case from the screenshot above.

> Could you try to set the href to the download link and extend your linked click handler to navigate to the preview page when ctrl is not clicked?

What would be the point? Making a change is fine only if there's value to do it and I still don't see any value in doing that

**Comment by darkdragon-001** [comment]
If you would have had a look at my screenshot, you noticed that there are actually two `click` handlers. Furthermore, changing `href` to something else still redirects to the _view_ URL on click. So there must be some other javascript event handler.

This is what I described in the ticket description: One could copy the download URL via right click menu directly in order to use it for wget/curl or other tools to download.

I am wondering why I take the time to add feature requests and add comments when you don't even read them. Wondering how you expect anyone to fund your project via opencollective or similar...

**Comment by mickael-kerjean** [comment]
> If you would have had a look at my screenshot .... I am wondering why I take the time to add feature requests and add comments ..... Wondering how you expect anyone to fund your project via opencollective or similar...

You might not have realised but I have spent a lot of time answering your messages already and I actually pay money each month to keep this project alive. 

Also you've gone too far with your comment, I don't want to interact with people like this in my free time, this is just too far out from why I'm doing open source and as such will remove everything you have opened, this is NOT ok.

**Comment by darkdragon-001** [comment]
> You might not have realised but I have spent a lot of time answering your messages...

This is what I am wondering as well: Most of my feature requests are pretty small (apart from #313 and #318 which I actually implemented myself in #321) and fixing for someone who knows the code base would take way less time than writing novels... Also the fix for this one is only a few lines if one knows where to add them...

**Comment by darkdragon-001** [comment]
> You might not have realised but I have spent a lot of time answering your messages...

This is what I am wondering as well: Most of my feature requests are pretty small (apart from #314 and #318 which I actually implemented myself in #321) and fixing for someone who knows the code base would take way less time than writing novels... Also the fix for this one is only a few lines if one knows where to add them...

I planned further contributions, but you are just making everything way more complicated than it is. Furthermore, commenting that you see 0 use for basically all of my suggested features (e.g. https://github.com/mickael-kerjean/filestash/issues/315#issuecomment-701396421, https://github.com/mickael-kerjean/filestash/issues/317#issuecomment-701992711) while other people show their interest by adding a thumbs up (e.g. #315, #322, https://github.com/mickael-kerjean/filestash/issues/318#issuecomment-711290694), doesn't encourage new contributors. Sorry if I sounded disrespectful, this clearly wasn't my intention but contributing to this project doesn't make sense for me as you make everything just unnecessarily complicated and time-consuming.
